### PR TITLE
Replace DRONE_ with CI_ variables in pipeline steps

### PR DESCRIPTION
**Pull request by anbraten**
# Changes
Dropped support for `DRONE_*` environment variables in pipeline steps. Pipeline meta-data can be accessed with `CI_*` variables.
  - `CI_*` prefix replaces `DRONE_*`
  - `CI` value is now `woodpecker`
  - `DRONE=true` has been removed

# Follow ups
- add `CI_AGENT_*` variables
- improve agent (cli exec) arch detection
- improve tracer environment updating

### CODE REVIEW

File: pipeline/frontend/metadata.go [hunk 1]
```diff
@@ -18,22 +19,23 @@ const (
...
-		Prev Build  `json:"prev,omitempty"`
-		Job  Job    `json:"job,omitempty"`
-		Sys  System `json:"sys,omitempty"`
+		ID    string `json:"id,omitempty"`
+		Repo  Repo   `json:"repo,omitempty"`
+		Curr  Build  `json:"curr,omitempty"`
+		Prev  Build  `json:"prev,omitempty"`
+		Job   Job    `json:"job,omitempty"`
+		Sys   System `json:"sys,omitempty"`
+		Agent Agent  `json:"agent,omitempty"`
```
Comment (6543) [inline pipeline/frontend/metadata.go:28]: need own pull i would say

Comment (anbraten) [reply pipeline/frontend/metadata.go]: reverted those changes for now

File: docs/docs/91-migrations.md [hunk 2]
```diff
@@ -12,15 +12,36 @@ Some versions need some changes to the server configuration or the pipeline conf
...
-- From version `0.15.0` ongoing there will be three types of docker images: `latest`, `next` and `x.x.x` with an alpine variant for each type like `latest-alpine`.
-  If you used `latest` before to try pre-release features you should switch to `next` after this release.
+- Dropped support for `DRONE_*` environment variables in pipeline steps. Pipeline meta-data can be accessed with `CI_*` variables.
+  - `CI_*` prefix replaces `DRONE_*`
+  - `CI` value is now `woodpecker`
+  - `DRONE=true` has been removed
+  - Some variables got deprecated and will be removed in future versions. Please migrate to the new names. Same applies for `DRONE_` of them.
+    - CI_ARCH => use CI_SYSTEM_ARCH
+    - CI_COMMIT => CI_COMMIT_SHA
+    - CI_REMOTE_URL => use CI_REPO_REMOTE
```
Comment (6543) [inline docs/docs/91-migrations.md:32]: as i understand the pull we still have no DRONE_ fallback

that would make all existing plugins using them useless!

Comment (6543) [reply docs/docs/91-migrations.md:32]: we dont have to promote them and I would add them in a seperate func (even seperated in a own go file) but It should be there

File: pipeline/pipeline.go [hunk 3]
```diff
@@ -127,6 +128,13 @@ func (r *Runtime) exec(proc *backend.Step) error {
 		}
 	}
 
+	// TODO: using DRONE_ will be deprecated with 0.15.0. remove fallback with following release
```
Comment (anbraten) [inline pipeline/pipeline.go:131]: @6543 This is now generating the `DRONE_` environment variables based on the `CI_` versions by simply replacing the prefixes. This way seemed to be much easier to maintain IMO.

Comment (anbraten) [reply pipeline/pipeline.go:131]: I placed it directly before executing the pipeline step, so every environment variable will be caught (so it works for cli this way as well).

Comment (6543) [reply pipeline/pipeline.go:131]: oh! - ok I'll review tomorow

File: agent/runner.go [hunk 4]
```diff
@@ -276,26 +276,18 @@ func (r *Runner) Run(ctx context.Context) error {
 			state.Pipeline.Step.Environment = map[string]string{}
 		}
 
-		state.Pipeline.Step.Environment["DRONE_MACHINE"] = r.hostname
+		state.Pipeline.Step.Environment["CI_MACHINE"] = r.hostname
 		state.Pipeline.Step.Environment["CI_BUILD_STATUS"] = "success"
 		state.Pipeline.Step.Environment["CI_BUILD_STARTED"] = strconv.FormatInt(state.Pipeline.Time, 10)
 		state.Pipeline.Step.Environment["CI_BUILD_FINISHED"] = strconv.FormatInt(time.Now().Unix(), 10)
-		state.Pipeline.Step.Environment["DRONE_BUILD_STATUS"] = "success"
```
[the remark this answers is not in the document]
Comment (anbraten) [reply agent/runner.go:278]: FYI: From my research it seems that these `state....Environment[...]` variables are only used for internal state checks and have nothing to do with environment variables in the common sense (those changes are not applied to any containers as env vars). If I am correct we could refactor them to be a special state struct with variables like started, finished and status.

**Review by 6543 — APPROVED** [review]
well yes move on :)

**Pull request merged**